File: dimdwarf-core/src/main/java/net/orfjackal/dimdwarf/net/SimpleSgsProtocolCodecFactory.java
// Copyright © 2008-2010 Esko Luontola <www.orfjackal.net>
// This software is released under the Apache License 2.0.
// The license text is at http://dimdwarf.sourceforge.net/LICENSE

package net.orfjackal.dimdwarf.net;

import org.apache.mina.core.session.IoSession;
import org.apache.mina.filter.codec.*;

public class SimpleSgsProtocolCodecFactory implements ProtocolCodecFactory {

    public ProtocolEncoder getEncoder(IoSession session) throws Exception {
        return new SimpleSgsProtocolEncoder();
    }

    public ProtocolDecoder getDecoder(IoSession session) throws Exception {
        return new SimpleSgsProtocolDecoder();
    }
}

File: dimdwarf-core/src/main/scala/net/orfjackal/dimdwarf/net/sgs/SimpleSgsProtocolCodecFactory.scala
package net.orfjackal.dimdwarf.net.sgs

import org.apache.mina.core.session.IoSession
import org.apache.mina.filter.codec._

class SimpleSgsProtocolCodecFactory extends ProtocolCodecFactory {
  private val encoder = new SimpleSgsProtocolEncoder
  private val decoder = new SimpleSgsProtocolDecoder

  def getEncoder(session: IoSession): ProtocolEncoder = encoder

  def getDecoder(session: IoSession): ProtocolDecoder = decoder
}
